ff-fft/src/polynomial/dense.rs

```diff
@@ -104,11 +106,24 @@ impl<F: Field> DensePolynomial<F> {
         }
         assert_eq!(powers_of_point.len(), self.coeffs.len());
         let zero = F::zero();
-        powers_of_point
```

Comment:
Let's do something similar to here: https://github.com/scipr-lab/zexe/blob/8a9343ad03e24cf278598e19c9a7b928b3ab729e/algebra/src/msm/fixed_base.rs#L80

This avoid duplicating the rest of the iterator chain.

Reply:
in this case the iterator chain is not duplicated, cause `fold/reduce` function from `rayon` has a different `signature` than `fold` from `std` and it requires us to pass `Fn` as accumulator

[rayon-docs#method.fold](https://docs.rs/rayon/1.3.0/rayon/iter/trait.ParallelIterator.html#method.fold)

[std-docs#method.fold](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold)